src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -580,6 +580,21 @@ internal static bool HasUnsuppressedErrors(DiagnosticBag diagnostics)
             return false;
         }
 
+        /// <summary>
+        /// Returns true if there are any unsuppressed error diagnostics in the bag.
+        /// </summary>
+        private static bool HasReportedErrors(DiagnosticBag diagnostics)
```

Comment:
>HasReportedErrors [](start = 28, length = 17)

Would it make sense to rename HasUnsuppressedErrors (the helper above) to HasUnsuppressableError and this helper to HasUnsuppressedErrors?

Reply:
Yes, that sounds better. I will do the rename of this method and even the internal helpers on `Diagnostic`.

Reply:
I was able to move this method to a static local method inside the Emit API - that should be the only place where we care about bailing out early in presence of warnings escalated as errors.